apm-sniffer/optional-reporter-plugins/kafka-reporter-plugin/src/test/java/org/apache/skywalking/apm/agent/core/kafka/KafkaProducerManagerTest.java

```diff
@@ -54,6 +55,17 @@ public void testFormatTopicNameThenRegister() {
         assertEquals(KafkaReporterPluginConfig.Plugin.Kafka.TOPIC_METRICS, value);
     }
 
+    @Test
+    public void testSetPropertiesFromJsonConfig() {
+        KafkaProducerManager kafkaProducerManager = new KafkaProducerManager();
+        Properties properties = new Properties();
+
+        KafkaReporterPluginConfig.Plugin.Kafka.PRODUCER_CONFIG_JSON = "{\"batch.size\":32768}";
+        kafkaProducerManager.setPropertiesFromJsonConfig(properties);
+
+        assertEquals(properties.get("batch.size"), "32768");
+    }
+
```

Comment:
What does this UT prove? The codes are not shaded ever in the UT scope, right?

Reply:
Show an example of how a JSON configuration can be correctly set to properties.

Reply:
> What does this UT prove? The codes are not shaded ever in the UT scope, right?

right.

Reply:
So, the point of this PR is `TypeToken`, which cast the type accordingly.

But still `import com.google.common.reflect.TypeToken;` seems not from gson package. Why?

Reply:
> So, the point of this PR is `TypeToken`, which cast the type accordingly.
> 
> But still `import com.google.common.reflect.TypeToken;` seems not from gson package. Why?

I have used the wrong TypeToken out of two.